src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -39,129 +40,112 @@
     }
-  }
-
-  /**
-   * Distribute SampleUnits for a SampleUnitGroup.
-   *
-   * @param exercise CollectionExercise of which sampleUnitGroup is a member.
-   * @param sampleUnitGroup for which to distribute sample units.
-   */
-  private void distributeSampleUnits(
-      CollectionExercise exercise, ExerciseSampleUnitGroup sampleUnitGroup) throws CTPException {
-    List<ExerciseSampleUnit> sampleUnits = sampleUnitRepo.findBySampleUnitGroup(sampleUnitGroup);
-    List<SampleUnit> children = new ArrayList<>();
-    SampleUnitParent parent = null;
-    for (ExerciseSampleUnit sampleUnit : sampleUnits) {
-      String activeActionPlanId =
-          sampleUnitDistributorHelper.getActiveActionPlanId(
-              exercise.getExercisePK(),
-              sampleUnit.getSampleUnitType().name(),
-              exercise.getSurveyId());
-      if (sampleUnit.getSampleUnitType().isParent()) {
-        parent = sampleUnit.toSampleUnitParent(activeActionPlanId, exercise.getId());
-      } else {
-        SampleUnit child = sampleUnit.toSampleUnitChild(activeActionPlanId);
-        children.add(child);
-      }
-    }
-
-    if (parent == null) {
-      log.with("sample_unit_group_pk", sampleUnitGroup.getSampleUnitGroupPK())
-          .error("No Parent for SampleUnit in SampleUnitGroup");
+    if (sampleUnitGroups.isEmpty()) {
+      log.with("collection_exercise_id", exercise.getId())
+          .debug("No sample unit groups to distribute for exercise");
       return;
     }
 
-    if (!children.isEmpty()) {
-      parent.setSampleUnitChildren(new SampleUnitChildren(children));
-    }
+    // Catch errors distributing sample units so that only failing units are stopped
+    sampleUnitGroups.forEach(
+        sampleUnitGroup -> {
+          try {
+            distributeSampleUnitGroup(exercise, sampleUnitGroup);
+          } catch (CTPException ex) {
+            log.with("sampleUnitGroupPK", sampleUnitGroup.getSampleUnitGroupPK())
+                .error("Failed to distribute sample unit group", ex);
+          }
+        });
 
-    if (parent.getActionPlanId() == null) {
-      String message =
-          String.format(
-              "Action Plan Id is required for collectionExerciseId=%s and sampleUnitId=%s",
-              exercise.getId(), parent.getId());
-      log.with("collection_exercise_id", exercise.getId())
-          .with("sample_unit_id", parent.getId())
-          .error("Action Plan Id is required");
-      throw new CTPException(CTPException.Fault.VALIDATION_FAILED, message);
-    }
+    // Collection exercise will transition to LIVE/READY_FOR_LIVE
+    // if all sample units were distributed successfully
+    collectionExerciseTransitionState(exercise);
 
-    publishSampleUnit(sampleUnitGroup, parent);
+    try {
+      sampleDistributionListManager.deleteList(DISTRIBUTION_LIST_ID, true);
+    } catch (LockingException ex) {
+      log.error("Failed to release sampleDistributionListManager data", ex);
```

Comment:
how do we get a locking exception if we're not doing any locking? (clearly it's somewhere in the call stack... maybe... seems mad to catch at all. I hate catch, log and do nothing because it makes flakey systems

Reply:
`sampleDistributionListManager.deleteList` throws LockingException, if I don't catch it I get compilation failures